MaxText/train.py

```diff
@@ -356,9 +361,13 @@ def main(argv: Sequence[str]) -> None:
       stack_trace_to_cloud = config.stack_trace_to_cloud,
       stack_trace_interval_seconds = config.stack_trace_interval_seconds))
   diagnostic_config = diagnostic_configuration.DiagnosticConfig(debug_config)
+  job_start_time = datetime.datetime.now()
   with diagnostic.diagnose(diagnostic_config):
     train_loop(config)
-
+  job_end_time = datetime.datetime.now()
+  goodput.goodput_log_job_runtime(config, job_end_time - job_start_time)
```

Comment:
Also you really don't want to pass a config outside of MaxText into your generic interface! Pass what you need!